gitlabform/processors/project/hooks_processor.py

```diff
@@ -9,23 +9,26 @@ def __init__(self, gitlab: GitLab):
         super().__init__("hooks", gitlab)
 
     def _process_configuration(self, project_and_group: str, configuration: dict):
+        debug("Processing hooks...")
+        project = self.gl.projects.get(project_and_group)
+        hooks_list = project.hooks.list()
+
         for hook in sorted(configuration["hooks"]):
+            hook_id = next((h.id for h in hooks_list if h.url == hook), None)
             if configuration.get("hooks|" + hook + "|delete"):
-                hook_id = self.gitlab.get_hook_id(project_and_group, hook)
                 if hook_id:
                     debug("Deleting hook '%s'", hook)
-                    self.gitlab.delete_hook(project_and_group, hook_id)
+                    project.hooks.delete(hook_id)
                 else:
                     debug("Not deleting hook '%s', because it doesn't exist", hook)
             else:
-                hook_id = self.gitlab.get_hook_id(project_and_group, hook)
+                attr = {"url": hook}
+                attr.update(configuration["hooks"][hook])
                 if hook_id:
                     debug("Changing existing hook '%s'", hook)
-                    self.gitlab.put_hook(
-                        project_and_group, hook_id, hook, configuration["hooks"][hook]
-                    )
+                    changed_hook = project.hooks.update(hook_id, attr)
```

Comment:
I'm leaning the python-gitlab library too. Is there a `update` method? Could you share where you found it in their docs? Maybe I'm not looking at the right place.

Reply:
I saw it in a dream... I even invented the parameters - I'm almost (!!!) certain I saw it somewhere but I can only find "save" now. Casts a dark shadow on my test :(

Reply:
Haha... If this exists, that'd be very helpful for some of the other logics to be migrated to python-gitlab library. See the following discussion: https://github.com/gitlabform/gitlabform/issues/73#issuecomment-1791761118

Looks like the acceptance test passed though. So, maybe this method does exist in the library and I just don't know how to process their documentation 😄

Reply:
I remember going through various links from the gitlab docs and finding all these examples of how to use the library - that's how I have this idea in my brain that the first parameter is the id (I had even had it without the id at some point and indeed, the test failed).... I'll dig deeper.

Reply:
OK, I might not be insane after all:
https://python-gitlab.readthedocs.io/en/v1.12.1/api/gitlab.html?highlight=update#

Reply:
That looks like a very old version of python-gitlab library. Looks like in latest version it's also available here: https://python-gitlab.readthedocs.io/en/stable/api/gitlab.html#gitlab.mixins.UpdateMixin

Reply:
Reading the API usage docs below, it seems `save()` used when updating something. Would be nice if it the `update()` was also documented with example.

https://python-gitlab.readthedocs.io/en/stable/api-usage.html